richtext-ui/src/commonMain/kotlin/com/halilibo/richtext/ui/string/Text.kt

```diff
@@ -101,13 +100,32 @@ public fun RichTextScope.Text(
   }
 }
 
-public typealias MarkdownAnimationState = Int
+public data class MarkdownAnimationState(
+  val lastAnimationStartMs: Long = 0,
+) {
+  public fun addAnimation(renderOptions: RichTextRenderOptions): MarkdownAnimationState = copy(
+    lastAnimationStartMs = calculatedDelay(renderOptions) + System.currentTimeMillis()
```

Comment:
SystemClock.uptimeMillis() is better for animation timings because it's monotonic. currentTimeMillis can change if the clock syncs, time zone changes, daylight savings, etc.

Reply:
unfortunately these aren't android modules so we don't have access to `import android.os.SystemClock` - and either way, I think the odds of the clock changing mid-animation are low enough to shrug off.